### PR TITLE
Ignore SSH keys in Recycle Bin, fixes #1396

### LINKED ISSUE #1396: SSH Agent Add Trash Keys
SSH Agent add automatically keys of entries in trash folder with auto add on unlock.

## Expected Behavior

For all entries in trash folder not processing ssh agent auto add on unlock.

## Current Behavior

All entries in trash with a SSH Agent parameters (with auto add on unlock) are added to ssh agent.

## Possible Solution

Remove from trash the entry or uncheck auto add to agent on unlock.

## Steps to Reproduce (for bugs)

1. Create an entry with ssh agent + auto add to agent on unlock. 
2. Save db.
3. Try it (lock db, unlock, check if ssh key is load `ssh-add -L`)
4. Remove entry (entry go to trash folder). 
5. Save db.
6. Clean all ssh keys `ssh-add -D` => `All identities removed.`
7. Check `ssh-add -L` => `The agent has no identities.`
8. Lock db, Unlock db, check if key is load  `ssh-add -L`
9. Key in trash is loaded.

## Context

Remove SSH entry only in trash to conserve it.
## Debug Info

KeePassXC - Version 2.2.4
Révision : 243edda

Bibliothèques :
- Qt 5.10.0
- libgcrypt 1.8.2

Système d'exploitation : Arch Linux
Architecture CPU : x86_64
Kernel : linux 4.14.13-1-ARCH

Extensions activées :
- KeePassHTTP
- Auto-Type
- YubiKey
- SSH Agent
- Native messaging browser extension

### PR DESCRIPTION
**Pull request by hifi**
Fixes #1396.

## Description
Ignore Recycle Bin when searching for SSH keys.

## Types of changes

- ✅ Bug fix (non-breaking change which fixes an issue)

## Checklist:

- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**

### CODE REVIEW

File: src/sshagent/SSHAgent.cpp [hunk 1]
```diff
@@ -217,6 +217,9 @@ void SSHAgent::databaseModeChanged(DatabaseWidget::Mode mode)
     } else if (mode == DatabaseWidget::ViewMode && !m_keys.contains(uuid.toHex())) {
         for (Entry* e : widget->database()->rootGroup()->entriesRecursive()) {
 
+            if (e->group()->iconNumber() == Group::RecycleBinIconNumber)
```
Comment (TheZ3ro) [inline src/sshagent/SSHAgent.cpp]: can you please add `{` on this if? Thanks

**Comment by hifi** [comment]
Fixed formatting for the next if as well.

File: src/sshagent/SSHAgent.cpp [hunk 2]
```diff
@@ -217,8 +217,13 @@ void SSHAgent::databaseModeChanged(DatabaseWidget::Mode mode)
     } else if (mode == DatabaseWidget::ViewMode && !m_keys.contains(uuid.toHex())) {
         for (Entry* e : widget->database()->rootGroup()->entriesRecursive()) {
 
-            if (!e->attachments()->hasKey("KeeAgent.settings"))
+            if (e->group()->iconNumber() == Group::RecycleBinIconNumber) {
```
Comment (phoerious) [inline src/sshagent/SSHAgent.cpp]: This should be a check for `e->group() == db->metadata()->recycleBin()` (with a preceding check for `db->metadata()->db->recycleBinEnabled() && db->metadata()->recycleBin()`). Only checking the icon is not sufficient.

**Comment by hifi** [comment]
Changed the method to detect Recycle Bin and rebased.

**Pull request merged**